pip_api/_parse_requirements.py

```diff
@@ -463,8 +475,20 @@ def parse_requirements(
         lines_enum = _skip_regex(lines_enum, options)
 
         for lineno, line in lines_enum:
-            req: Optional[Union[requirements.Requirement, UnparsedRequirement]] = None
+            req: Optional[Union[Requirement, UnparsedRequirement]] = None
             known, _ = parser.parse_known_args(line.strip().split())
+
+            hashes_by_kind = defaultdict(list)
+            if known.hashes:
+                for hsh in known.hashes:
+                    kind, hsh = hsh.split(":", 1)
```

Comment:
Not sure if it's worth it, but we could also promote any `ValueError` that happens here (due to a malformed `--hash` option) into a `PipError`.